packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/explore/TablesViewer.tsx

```diff
@@ -21,22 +20,19 @@ import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from ".
 import { bufferToBigInt } from "../utils/bufferToBigInt";
 import { EditableTableCell } from "./EditableTableCell";
 
-type Props = {
-  table: string | undefined;
-};
-
-export function TablesViewer({ table: selectedTable }: Props) {
-  const [sorting, setSorting] = useState<SortingState>([]);
-  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
-  const [columnVisibility, setColumnVisibility] = useState<VisibilityState>({});
-  const [rowSelection, setRowSelection] = useState({});
-  const [globalFilter, setGlobalFilter] = useState("");
-  const [showAllColumns, setShowAllColumns] = useState(false);
+export function TablesViewer() {
+  const [selectedTableId] = useQueryState("tableId");
+  const [globalFilter, setGlobalFilter] = useQueryState("filter", parseAsString.withDefault(""));
+  const [showAllColumns, setShowAllColumns] = useQueryState("showAllColumns", parseAsBoolean.withDefault(false));
+  const [sorting, setSorting] = useQueryState(
+    "sort",
+    parseAsArrayOf(parseAsJson<ColumnSort>()).withDefault(useRef([]).current),
```

Comment:
does `parseAsJson` not support arrays? why do you need the outer `parseAsArrayOf`?

Reply:
also what's this `useRef([]).current` pattern for? why not just `[]`

if you need a stable ref, could lift it out of the component like `const emptyColumnSort = []`

Reply:
Yeah, I needed a stable array reference to avoid infinite re-render. Lifted out initial value outside the component, was not sure which way is cleaner.